addons/misra_9.py

```diff
@@ -311,8 +311,7 @@ def parseInitializer(self, root, token):
                 if self.ed and self.ed.isValue:
                     if not isDesignated and len(self.rootStack) > 0 and self.rootStack[-1][1] == self.root:
                         self.rootStack[-1][0].markStuctureViolation(self.token)
-
-                    if isFirstElement and self.token.str == '0' and self.token.next.str == '}':
+                    if isFirstElement and self.token.getKnownIntValue() == 0 and self.token.next.str == '}':
```

Comment:
I like this approach however I feel we should also check self.token.isInt.. I assume that it would be non-compliant for various expressions that results in value 0..